maas/resource_maas_logical_volume.go

```diff
@@ -0,0 +1,215 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMAASLogicalVolume() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMAASLogicalVolumeCreate,
+		ReadContext:   resourceMAASLogicalVolumeRead,
+		UpdateContext: resourceMAASLogicalVolumeUpdate,
+		DeleteContext: resourceMAASLogicalVolumeDelete,
+
+		Schema: map[string]*schema.Schema{
+			"fs_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The file system type (e.g. `ext4`). If this is not set, the volume is unformatted.",
+			},
+			"machine": {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
Could be debatable but looking at `maas_block_device` this I would perhaps expect this to be the case if the idea of updating to a new machine doesn't make sense
```suggestion
				Required:    true,
				ForceNew:    true,
```

Reply:
This seems to reflect in your update function I think

Reply:
Hmm yeah, that's a good point